src/classes/app.py

```diff
@@ -94,6 +105,9 @@ def __init__(self, *args, mode=None):
             log.info("pyqt5 version: %s" % PYQT_VERSION_STR)
         except:
             pass
+			
+        # Connect QCoreApplication::aboutToQuit() signal to log end of the session
+        self.aboutToQuit.connect(self.onLogTheEnd)
```

Comment:
Smart. I wonder if more of the stuff in our `closeEvent()` handler shouldn't _actually_ be in a handler for the `aboutToQuit` signal? But, that's a side question.